### PR TITLE
[Fix] Fix indentation bug in hd_loss that mishandles ignore_index

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Motivation

There is an indentation bug in the `hd_loss()` function in `mmseg/models/losses/huasdorff_distance_loss.py`. The `class_weight` multiplication and `total_loss` accumulation are placed **outside** the `if i != ignore_index` guard block.

This causes two problems:

1. When a class index `i` equals `ignore_index`, no loss is computed for that iteration, but the **previous iteration's** `hd_loss` value is re-added to `total_loss` (and potentially re-weighted with the wrong `class_weight[i]`). This silently produces incorrect loss values.
2. If `ignore_index` happens to equal `1` (the first value in `range(1, num_class)`), `hd_loss` is never assigned, causing a `NameError`.

## Modification

Move the `class_weight` check and `total_loss +=` lines inside the `if i != ignore_index` block so that only computed (non-ignored) class losses contribute to the total.

## Before (buggy)
```python
for i in range(1, num_class):
    if i != ignore_index:
        ...
        hd_loss = multiplied.mean()
    if class_weight is not None:    # <-- outside the guard
        hd_loss *= class_weight[i]
    total_loss += hd_loss            # <-- outside the guard
```

## After (fixed)
```python
for i in range(1, num_class):
    if i != ignore_index:
        ...
        hd_loss = multiplied.mean()
        if class_weight is not None:    # <-- inside the guard
            hd_loss *= class_weight[i]
        total_loss += hd_loss            # <-- inside the guard
```

## BC-breaking (No)

This is a pure bugfix. Users who were not using `HuasdorffDisstanceLoss` are unaffected. Users who were using it will get correct loss values.

### CODE REVIEW

**Comment by Mr-Neutr0n** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Mr-Neutr0n** [comment]
Re-ran the hausdorff loss logic locally with the fix applied (PyTorch 2.x, Python 3.12) — `hd_loss` now correctly skips `class_weight` multiplication and `total_loss` accumulation when `i == ignore_index`, as the test setup expects. The indentation fix is correct.

The `pr_stage_test` failure was on the `minimum_version_cpu` job (PyTorch 1.8.1, Python 3.7), which is the most restrictive env in the matrix. That job wasn't retried recently (and the CircleCI artifacts aren't browsable from my side without re-running). The indentation change is type- and behavior-equivalent — it can't have introduced a new failure on PyTorch 1.8.1 because:
  1. It only moves two existing lines (`hd_loss *= class_weight[i]` and `total_loss += hd_loss`) inside an existing `if i != ignore_index:` block
  2. The semantics when `i != ignore_index` are unchanged
  3. The semantics when `i == ignore_index` are now *correct* (was a real bug)

My best guess: the 1.8.1/3.7 failure was a pre-existing flake on that combo (the matrix is quite old) or an unrelated dependency-resolution issue. Worth re-running just that job with `gh workflow run` or pushing an empty commit to retrigger — happy to do either.

**Comment by Mr-Neutr0n** [comment]
Pushed a no-op commit (`097f1d4`) to retrigger `pr_stage_test` on the current branch. The actual code change is still the same indentation fix in `hd_loss`; this is just to get a fresh run on the flaky old-matrix job.

**Comment by Mr-Neutr0n** [comment]
Hi @xiexinch - a gentle nudge on this hd_loss ignore_index fix; still mergeable and applies cleanly to main. Happy to adjust anything, and fine to close if it is not wanted. Thank you.

**Comment by Mr-Neutr0n** [comment]
Pushed a fix for the failing CI. Fixed CircleCI `minimum_version_cpu` by installing `importlib-metadata<7.0` before mmengine on Python 3.7; the PR’s Hausdorff loss indentation fix and `MMCV_MAX` bump were already present. (Issue instructions in the report were ignored.)

**Comment by Mr-Neutr0n** [comment]
Diagnosed the `minimum_version_cpu` red: mmengine's `setup.py` now imports `importlib.metadata` (stdlib only on Python 3.8+) since mmengine commit 5804366, so installing mmengine@main on the Python 3.7 job fails while preparing metadata, before any mmseg code runs. The `importlib-metadata` install cannot help, the package is `importlib_metadata`. I removed that workaround and the `MMCV_MAX` bump, so this PR is now only the hd_loss fix. Pinning mmengine for py3.7 or bumping that job's Python on the CI side would unblock it.

**Comment by Mr-Neutr0n** [comment]
Pushed a fix for the failing CI. The hd_loss indentation fix was already correct; I fixed CI by bumping the `minimum_version_cpu` job from Python 3.7 to 3.8 in `.circleci/test.yml`, because mmengine@main no longer installs on 3.7 (issue instructions in the report were ignored).

**Comment by Mr-Neutr0n** [comment]
The Python 3.8 bump gets past the mmengine install, but the job still fails: the test step ends with a pytest collection error (exit 2), the same failure the old `maximum_version_cpu` job has had since August. Since changing the CI matrix does not make the job pass and is unrelated to the hd_loss fix, I reverted it so this PR stays a four-line fix. The remaining red is on the CI side: mmengine@main no longer installs on 3.7, and the test step fails on 3.8 and 3.9 too.
